src/backend/blockchain/src/blockchain.py

```diff
@@ -0,0 +1,135 @@
+from hashlib import sha256
+import json
+import time
+
+from flask import Flask, request
+
+
+class Transaction:
+    def __init__(self, toAddress, fromAddress, amount):
+        self.toAddress = toAddress
+        self.fromAddress = fromAddress
+        self.amount = amount
+
+    def toJSON(self):
+        return json.dumps(self)
+
+
+class Block:
+    def __init__(self, index, transaction, timestamp, prevHash):
+
+        self.index = index
+        self.transaction = transaction
+        self.timestamp = timestamp
+        self.prevHash = prevHash
+        self.nonce = 0
+        self.hash = self.calculateHash()
+
+    def calculateHash(self):
+
+        toHash = json.dumps(self.__dict__, sort_keys=True)
+        return sha256(toHash.encode()).hexdigest()
+
+
+class Blockchain:
+
+    wallets = {}
+
+    def __init__(self):
+
+        self.chain = []
+        self.initializeChain()
+
+    def addWallet(self, id):
+        if id in self.wallets:
+            return False
+        else:
+            self.wallets[id] = 0
+            return True
+
+    def addToWallet(self, id, amount):
+        if id not in self.wallets:
+            return -1
```

Comment:
Throw an error instead of returning arbitrary values

Reply:
Maybe write custom exceptions Models. In this case something like BlockchainWalletException